lib/renderer.js

```diff
@@ -1,21 +1,31 @@
 'use strict';
 
 const less = require('less');
-const { basename, dirname, join} = require('path');
+const { basename, dirname, join } = require('path');
+const micromatch = require('micromatch');
 
 module.exports = async function(data) {
   this.theme.config.less = Object.assign({
     paths: []
   }, this.theme.config.less);
+  const { route, theme } = this;
 
-  const config = this.theme.config.less;
+  const config = theme.config.less;
   const cwd = process.cwd();
-  const paths = config.paths.map(filepath => {
-    return join(cwd, filepath); // assuming paths are relative from the root of the project
-  });
+  const routeList = route.list();
+  let tmpPaths = config.paths;
+
+  if (!tmpPaths) tmpPaths = [];
+  else if (typeof tmpPaths === 'string') tmpPaths = [tmpPaths];
+
+  const paths = tmpPaths.filter(path => !micromatch.scan(path).isGlob)
+    .map(path => join(cwd, path)); // assuming paths are relative from the root of the project;
+  const glob = tmpPaths.filter(path => micromatch.scan(path).isGlob);
+  const match = micromatch(routeList, glob).map(path => join(cwd, dirname(path)));
```

Comment:
> Any ideas about non-glob support?

forgot about the non-glob. In retrospect, I don't think it's necessary to filter out glob patterns, can pass the config `config.paths` directly to micromatch.